storage/innobase/log/log0log.cc

```diff
@@ -1385,20 +1355,40 @@ redo log file header.
 @return true if success. */
 bool
 log_rotate_encryption() {
-	fil_space_t* space = fil_space_get(SRV_LOG_SPACE_FIRST_ID);
- 	if (!FSP_FLAGS_GET_ENCRYPTION(space->flags)) {
+	fil_space_t *space = fil_space_get(SRV_LOG_SPACE_FIRST_ID);
+	if (!FSP_FLAGS_GET_ENCRYPTION(space->flags)) {
 		return(true);
 	}
- 	/* Rotate log tablespace */
-	return(log_write_encryption(NULL, NULL, static_cast<redo_log_encrypt_enum>(srv_redo_log_encrypt)));
+	/* Rotate log tablespace */
+	return (log_write_encryption(
+	    NULL, NULL,
+	    static_cast<redo_log_encrypt_enum>(srv_redo_log_encrypt)));
+}
+
+void
+log_check_new_key_version() {
+	if (srv_shutdown_state != SRV_SHUTDOWN_NONE) {
+		return;
+	}
+	fil_space_t* space = fil_space_get(SRV_LOG_SPACE_FIRST_ID);
+	if (!FSP_FLAGS_GET_ENCRYPTION(space->flags)) {
+		return;
+	}
+	if (srv_redo_log_encrypt == REDO_LOG_ENCRYPT_RK) {
+		/* re-fetch latest key */
+		redo_log_key* mkey = redo_log_key_mgr.load_latest_key();
+		if (mkey != NULL) {
+			space->encryption_redo_key = mkey;
+			srv_redo_log_key_version = mkey->version;
+		}
+	}
 }
 
 /** Check the redo log encryption is enabled or not.
 It will try to enable the redo log encryption and write the metadata to
 redo log file header. */
 void
-redo_rotate_default_key()
-{
+redo_rotate_default_key() {
```

Comment:
minor request: only this function name convention looks different. Can we change to log_rotate_default_key(). (I know it exists before too)